DoSomething/gambit
# Chatbot testing

#### What's this PR do?
I have started to refactor middleware in the chatbot route. It adds test coverage for these middleware. It also has some refactoring.

#### How should this be reviewed?
- 👀 
- Run `npm run all-tests`
- Use postman to signup and reportback to a bot campaign. It should work with no problemo.

#### Any background context you want to provide?
A lot of movement in the code.

#### Checklist
- [x ] Tested on localhost by signing up and reporting back to SEEDBOT and LUCKYBOT.
